src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1316,18 +1318,22 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
 
             // Dev11 searches for the key file in the current directory and assembly output directory.
             // We always look to base directory and then examine the search paths.
-            if (!string.IsNullOrEmpty(baseDirectory))
+            if (!RoslynString.IsNullOrEmpty(baseDirectory))
             {
                 keyFileSearchPaths.Add(baseDirectory);
             }
 
-            if (!string.IsNullOrEmpty(outputDirectory) && baseDirectory != outputDirectory)
+            if (RoslynString.IsNullOrEmpty(outputDirectory))
+            {
+                AddDiagnostic(diagnostics, ErrorCode.ERR_NoOutputDirectory);
```

Comment:
:memo: this new error guards against `null` output directory, allowing us to declare `OutputDirectory` as non-nullable.

Reply:
📝 This new error means loc change, and the cutoff for 16.5 already passed so this needs to retarget 16.6 Preview 1

Reply:
What was the previous behavior?

---
In reply to: [370462374](https://github.com/dotnet/roslyn/pull/41063#discussion_r370462374) [](ancestors = 370462374)

Reply:
You could get a null OutputDirectory, which we're not prepared to handle in code that reads that property.
I'm not sure if that emitting code (which reads the property) could be reached though.

---
In reply to: [377830798](https://github.com/dotnet/roslyn/pull/41063#discussion_r377830798) [](ancestors = 377830798,370462374)